pages/blog/_posts/2020-05-18-formatting-erlang.md

```diff
@@ -0,0 +1,321 @@
+---
+layout: blogpost
+permalink: /blog/formatting-erlang/
+title: Formatting Erlang
+date: 2020-05-18
+tags: erlang rebar3 coding culture
+author: <a href="https://www.linkedin.com/in/sebastian-weddmark-olsson/">Sebastian Weddmark Olsson</a>
+---
+
+I've looked into different formatters for Erlang and there are some
+options. The main two alternatives being `steamroller` and
+`rebar3_format`. `steamroller` is slow, which one could have imageined
+by its name. `rebar3_format` have issues with macros due to underlying
+dependencies.
+
+I've also briefly looked into an Erlang linter called `elvis`.
+
+# Background
+
+In Working Group Two we use a bunch of different programming
+languages, and we all have different experiences and are used to
+different languages and environments. We are pretty autonomous and we
+are expected to jump in and out in different services to fix bugs and
+add features.
+
+We like when the code is uniform because that make it easier to
+understand it. That is why we want to use tools to make sure our code
+is consistent no matter who writes it, which tool, or regards to
+language rules or background.
+
+About half a year ago there was a discussion about code style within
+WG2 that resulted in formatting tools being applied for Kotlin, Bazel,
```

Comment:
We agreed to always use `Working Group Two` (the full name) when referring to ourselves.